callbacks/agent_final_stream_test.go

```diff
@@ -0,0 +1,37 @@
+package callbacks
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type filterFinalStringTestCase struct {
+	inputStr string
+}
+
+func TestFilterFinalString(t *testing.T) {
+	t.Parallel()
+
+	keywork := "Final Answer:"
+
+	// correct final string
+	correctStr := "This is a correct final string."
+
+	extraStrAbore := fmt.Sprintf(" some other text.\nFinal Answer: %s", correctStr)
+	extraStrBefore := fmt.Sprintf(" another text. Final Answer: %s", correctStr)
+	extraColonWithSpacesBefore := fmt.Sprintf(`   :    %s`, correctStr)
+
+	testCases := []filterFinalStringTestCase{
```

Comment:
please place this at the top of the test (after t.Parallel()) and approach it like:

```go
cases := []struct { ...```

Reply:
updated all test case inline with:
```go
cases := []struct { ...
```
please review